jms/1.7/modules/ROOT/pages/jms-listener.adoc

```diff
@@ -1,43 +1,76 @@
 
+== Listen for New Messages from a Queue
 
-The On New Message (Listener) source in the JMS connector provides ability to consume Messages as they arrive to the Destination.
+In the following example, you configure the *On New Message* source to listen for new messages in the queue specified in the *Destination* field. The operation returns a Mule Message each time a JMS message is available in the queue. The Mule Message has:
 
-== Listening for New Messages
-The syntax to listen for new messages from a queue is:
+* The message's content as payload
+* The message's metadata in the message attributes
+
+By default, the message consumed will be acknowledged (ACKed) only when the execution of the flow receiving the message completes successfully. If instead, an error occurs during the execution of the flow, the session recovers, and the message is redelivered.
+
+To configure the source in Studio, follow these steps:
+
+. In the *Mule Palette* view, select *JMS > On New Message*.
+. Drag *On New Message* to the Studio canvas.
+. In the *On New Message* configuration screen, optionally change the value of the *Display Name* field.
+. In the *On New Message* configuration screen, in *Destination*, specify the name of the destination from where to consume the message for example `#[vars.destination]`.
+
+.JMS Connector On New Message on Queue configuration
+image::jms-onnewmessage-queue.png[In the On New Message source screen, set the Destination field to the name of the destination from where to consume the message]
+
+In the XML editor, the `<jms:listener>` configuration looks like this:
 
 [source,xml,linenums]
 ----
 <jms:listener config-ref="config" destination="#[vars.destination]"/>
 ----
 
-The source above listens for new messages in the Queue identified by the destination, returning a `MuleMessage` each time a JMS Message is available in the Queue.
+== Listen for New Messages from a Topic Subscription
 
-The _MuleMessage_ will have:
+In the following example, you configure the *On New Message* source to listen for new messages from a topic subscription by setting the *Consumer type* field to *Topic consumer*.
 
-* The message's content as payload.
-* The message's metadata in the message attributes.
+To configure the source in Studio, follow these steps:
 
-By default, the Message consumed will be ACKed only when the execution of the flow receiving the message completes successfully.
-If instead, an error occurs during the execution of the flow, the Session will be recovered and the Message will be redelivered.
+. In the *Mule Palette* view, select *JMS > On New Message*.
+. Drag *On New Message* to the Studio canvas.
+. In the *On New Message* configuration screen, optionally change the value of the *Display Name* field.
+. In the *On New Message* configuration screen, in *Destination*, specify the name of the destination from where to consume the message for example `#[vars.destination]`.
```

Comment:
```suggestion
. In the *On New Message* configuration screen, in *Destination*, specify the name of the destination from which to consume the message for example `#[vars.destination]`.
```

Reply:
I actually looked this up, because I had never heard "in where" used. It seems to be either something that sounds better to a native US English-speaker's ear, or "in where" is never used but "where" (vs. "in which") is.
It might be easiest just to take my word for it...lol....